### PR TITLE
Rename `true_div` to `true_divide` to match with NumPy

### PR DESCRIPTION
**Pull request by sudarsan2k5**
Addressing #1212 

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [x] There is an informative high-level description of the changes.
+ [x] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [x] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [x] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [x] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [x] There are tests covering the changes introduced in the PR.

### CODE REVIEW

**Comment by rlouf** [comment]
Thank you for contributing! I think we'll need to alias `true_div` to not break the code that currently relies on this, and add a deprecation warning.

**Comment by sudarsan2k5** [comment]
Hi @rlouf I have added an alias for `true_division` but could you please guide me how to set deprecation warning for an alias? I know how to do with Function but not sure for an alias/variable.

**Comment by rlouf** [comment]
The alias is a function, and you can add it to the `DEPRECATED_NAMES` list at the end of the module.

**Comment by sudarsan2k5** [comment]
Hi @rlouf, as you suggested, I have created an alias `divide` to `true_divide` and deprecated `true_div`.

Could you please check that I did everything correctly because there is no deprecation warning while importing it?

Is there anything else we need to do for `true_div` in this PR? Also, I'm not sure how to fix the failed cases; please advise.

Is it also necessary to rename the class `TrueDiv` to `TrueDivide`?

File: aesara/scalar/basic.py [hunk 1]
```diff
@@ -4455,6 +4458,11 @@ def handle_composite(node, mapping):
     ("Inv", "`Inv` is deprecated; use `Reciprocal` instead.", Reciprocal),
     ("inv", "`inv` is deprecated; use `reciprocal` instead.", reciprocal),
     ("Scalar", "`Scalar` is deprecated; use `ScalarType` instead.", ScalarType),
+    (
+        "true_div",
+        "`true_div` is deprecated; use `true_divide` or `divide` instead.",
+        true_div,
```
Comment (brandonwillard) [inline aesara/scalar/basic.py:4464]:
```suggestion
        true_divide,
```

File: aesara/scalar/basic.py [hunk 2]
```diff
@@ -793,8 +793,8 @@ def __sub__(self, other):
     def __mul__(self, other):
         return mul(self, other)
 
-    def __truediv__(self, other):
-        return true_div(self, other)
+    def __truedivide__(self, other):
```
Comment (brandonwillard) [inline aesara/scalar/basic.py:796]: Is `__truedivide__` an actual Python Dundee method?

**Review by brandonwillard — CHANGES_REQUESTED** [review]

File: aesara/scalar/basic.py (hunk 2 above)
Comment (sudarsan2k5) [reply aesara/scalar/basic.py:796]: Yes, I checked again and found [`__truediv__`](https://docs.python.org/3/library/operator.html#operator.__truediv__) is a python operator so I have a correct this, Thanks for correcting me.

Comment (brandonwillard) [reply aesara/scalar/basic.py]: Ha, I like that spell check on my phone changed "dunder" to "Dundee"!

Comment (brandonwillard) [reply aesara/scalar/basic.py]: Seems like we need some kind of meme GIF for [this situation](https://knowyourmeme.com/memes/thats-a-knife).

Comment (sudarsan2k5) [reply aesara/scalar/basic.py]: ha ha ha 🤣

**Review by brandonwillard — APPROVED** [review]

**Comment by brandonwillard** [comment]
Thanks @sudarsan2k5!

**Pull request merged**